Survey/Sources/Data/NetworkAPI/Interceptors/AuthenticationInterceptor.swift

```diff
@@ -30,4 +35,61 @@ final class AuthenticationInterceptor: RequestInterceptor {
             completion(.failure(NetworkAPIError.generic))
         }
     }
+
+    func retry(
+        _ request: Alamofire.Request,
+        for session: Session,
+        dueTo error: Error,
+        completion: @escaping (RetryResult) -> Void
+    ) {
+        guard request.retryCount == 0 else { return completion(.doNotRetry) }
+
+        if refreshRequest != nil { return }
+
+        refreshRequest = getRefreshToken { [weak self] result in
+            switch result {
+            case let .success(authToken):
+                try? self?.keychain.set(newToken: KeychainToken(token: authToken), for: .token)
```

Comment:
Since this is marked with `try` and keychain needs this set to success, should we try catch this and return error if `set` is not success?

Reply:
good suggestion 👍 , I updated at https://github.com/nkhanh44/ic-survey-ios/pull/55/commits/b85a01436ba1a05779591a14c97fb15abd40d103